spec/lib/export_tables_to_cloud_storage_spec.rb

```diff
@@ -0,0 +1,32 @@
+require 'rails_helper'
+require 'export_tables_to_cloud_storage'
+require 'fileutils'
+
+RSpec.describe ExportTablesToCloudStorage do
+  before do
+    FileUtils.rm_rf(Rails.root.join('tmp/csv_export'))
```

Comment:
Don't worry about interacting with the filesystem, just mock it out in exactly the same way you've done with the GCP buckets.

Probably something like `expect (fileutil_mock).to receive(:rm_rf).with('tmp/csv_export')`.  It'll remove the need to clear existing temp files and make the tests faster.